doc/glossary.rst

```diff
@@ -307,6 +307,17 @@
                 state of a multi-site configuration. When the period is updated,
                 the "epoch" is said thereby to have been changed.
 
+        Placement Groups (PGs)
```

Comment:
We really didn't have an entry for this already?  Sheesh.

Suggested wording that adjust a number of nuances:

>Placement groups (PGs) are subsets of each logical Ceph pool.
They perform the function of placing objects
(as a group) into OSDs. Ceph manages data internally at placement group
granularity: this scales better than managing individual and much more
numerous RADOS objects. A cluster that provisions an appropriate number of
placement groups (e.g., 100 per OSD) is better able to balance data distribution
than an otherwise identical cluster with a smaller number of placement groups.

Ceph's internal RADOS objects are each mapped to a specific placement group,
and each placement group belongs to exactly one Ceph pool.

Reply:
"Sheesh" is basically what I said, too.

I have incorporated your changes (and made one small refinement that I think makes a sentence a little more syntactically correct). 

Sheesh no more!

Reply:
I did word the balanced part deliberately, but the nuance isn't critical.  While as you revised is accurate, it's not as precise.  CRUSH distribution is in Sage's words "probabilistic", so there is always a bell curve.  More PGs naturally can result in a smoother curve.  One typically uses one of several "balancer" methods to refine/correct that curve, squeezing it so that the standard deviation is much smaller, and even more importantly so that the most-full outlier OSD's fullness is within a couple of percent of the average.  I tend to use "fullness" in this context as other terms can be ambiguous.

Before ... Luminous? there was no bundled balancer, today we have a ceph-mgr module that's enabled by default.

Previous strategies included manual or automated iterations of `ceph osd reweight-by-utilization`.

There is also the "JJ balancer" that works better with certain topologies.

Without any balancer the bell curve will manifest.  Since the mgr module has been enabled by default for several releases we can safely assume that *some* balancer is in effect, so the above discourse is for context not for any suggested change to this text.  Which of course is a glossary 👅 not a tutorial.

Reply:
Zac of the future: unfinished business